src/cmake/thirdparty/FindLUA.cmake

```diff
@@ -0,0 +1,55 @@
+# Copyright (c) 2017-2020, Lawrence Livermore National Security, LLC and
+# other Axom Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (BSD-3-Clause)
+#------------------------------------------------------------------------------
+# Setup Lua
+#------------------------------------------------------------------------------
+# This file defines:
+#  LUA_FOUND - If Lua was found
+#  LUA_INCLUDE_DIRS - The Lua include directories
+#  LUA_LIBRARY - The Lua library
+#------------------------------------------------------------------------------
+
+# first Check for LUA_DIR
+if (NOT EXISTS "${LUA_DIR}")
+    message(FATAL_ERROR "Given LUA_DIR does not exist: ${LUA_DIR}")
+endif()
+
+if (NOT IS_DIRECTORY "${LUA_DIR}")
+    message(FATAL_ERROR "Given LUA_DIR is not a directory: ${LUA_DIR}")
+endif()
```

Comment:
For this type of sanity checks, you may do:
```
blt_assert_exists( DIRECTORIES ${LUA_DIR} 
                               ${LUA_DIR}/include
                               ${LUA_DIR}/include/lua
                               ${LUA_DIR}/lib

                    FILES      ${LUA_DIR}/include/lua.hpp )
```

It was these kind of sanity checks that I had in mind that inspired creation of that blt macro....